eng/generators.targets

```diff
@@ -3,6 +3,8 @@
   <PropertyGroup>
     <EnableLibraryImportGenerator Condition="'$(EnableLibraryImportGenerator)' == '' and
                                              '$(MSBuildProjectName)' == 'System.Private.CoreLib'">true</EnableLibraryImportGenerator>
+
+    <ShouldUnsetParentConfigurationAndPlatform>false</ShouldUnsetParentConfigurationAndPlatform>
```

Comment:
I'm not sure if it's OK to set this globally. And if there's a way to only do this per project reference (which would be cleaner in this case).
I plan to add a comment to describe why this is done if we decide to leave it here as-is.

Reply:
cc @ViktorHofer 

I don't think there's actually a more targeted way so make it stop clearing out Configuration, since the [`AssignProjectConfiguration` task](https://github.com/dotnet/msbuild/blob/3c910ba83fc9dbd8e12f50dddc8c381404f928c4/src/Tasks/AssignProjectConfiguration.cs#L211-L228) uses that property. I think the global property makes sense - it also gives us consistency in behaviour between building in VS and on the command line. 

We actually already set it for projects under libraries: https://github.com/dotnet/runtime/blob/e21a975403d1d8b2b0432faf3e3286fe4196f965/src/libraries/Directory.Build.props#L15
I wonder if it would actually make sense to set it at the repo level?

Reply:
Here's the only documentation that I could find for that property: https://github.com/dotnet/msbuild/blob/d1af34332573aab3de6482ac1d0b35e1d7951564/src/Tasks/AssignProjectConfiguration.cs#L118-L123

I don't remember why this was set in libraries, but it had to do with the TargetFramework configuration system bring-up in runtime. Anirudh implemented that. @ericstj do you remember why this had to be set?

Reply:
I don't remember it, but here's the PR that added it: https://github.com/dotnet/runtime/pull/32259

My best guess is that SLNs act as an indirection that are meant to map every Configuration|Platform tuple for the sln to some Configuration|Platform tuple for the referenced project.  Maybe our SLNs were missing release sections, or maybe missing sections for the entire closure of references?  At that time we were not mucking with Configuration any more (we switched to TargetFramework) so I'm not entirely sure if that's related to targeting infra.